Create json for EQIP IRA

ywkim312 opened the pull request:
There two outputs one for summary and the other for state distribution. The result json "eqip_ira_state_distribution.json" and "eqip_ira_summary.json" are the ones that you want to check. For state distribution, the parser combines three tables. The main one will be the one that contains the data for 2023 and total. Then there are two additional table for future min and max values. I added these additional tables' future years' values to corresponding practice name of the 2023/total of each corresponding state in the output json. So some of the practices doesn't have any future data when they don't have any matching future min/max values. For summary, it only uses the first main table not the future min/max table

I am not very sure if the format is correct, so any suggestion will be welcomed.

ywkim312 committed:
embed EQIP IRA to proper branch (#98)

* Renamed Pastured cropland to Grassland in CSP data

* Updated changelog

* State distribution json for EQIP IRA

ywkim312 committed:
modified changelog

ywkim312 committed:
added summary

pengyin-shan commented on title-2-conservation/eqip_ira/eqip_ira_state_distribution.json:
```diff
@@ -0,0 +1,16975 @@
+        "state": "TX",
+        "practices": [
+            {
+                "practiceName": "Total",
```
Quick question, I was referring our current title II setup
<img width="797" alt="Screenshot 2024-05-17 at 3 40 40 PM" src="https://github.com/policy-design-lab/data-import/assets/92752107/98517010-9c1d-4d45-92fa-fcbe5d69bafc">
We make the totalPayment as the same level as the state name. Should we follow the same pattern?

ywkim312 replied:
It only has 2023 data and future data is not complete. Some of them has values and some doesn't hmmm maybe we want discuss about that. Also there is no statute either. All of them are EQIP IRA program and it has practices under it.

pengyin-shan replied:
Sorry if I miscommunicated. For IRA, we have a `"practiceName": "Total"... `JSON item, while in the regular data, we just use `totalPaymentInDollars` instead. Can we do the same? Or did you mean that not all practices have a total?

It's okay that we don't have 'status'; I will look for 'practices' instead, as you have already set up.

ywkim312 replied:
There is a state name as 'total' in the table and 'total' for each practice name. And I don't know what it means exactly and my assumption that is tit is the total for all the state. I strongly suggest you to check the 20240215_EQIP_IRA.csv file

pengyin-shan replied:
Interesting...@sandeep-ps do you have any clue about the 'total' means? Should we seek clarification with Johnathan?

sandeep-ps replied:
Sorry, I missed this message/email notification earlier. I think this state name with the value "Total" can be ignored, as we can calculate the total while generating the eqip_ira_summary.json. If possible, please include `totalPracticeInstanceCount` and `totalPaymentInDollars` at the root level of eqip_ira_summary.json.

We also have a practice name with the value "Total", which needs to be clarified with Yifan. I will send an email and CC you both. Thanks.

sandeep-ps replied:
Based on feedback from Yifan via emails (in which you are copied), it looks like we can remove both the state name "Total" and practice name "total" from different state data in the JSON file. 

As I suggested in my previous comment in this thread, if we can include `totalPracticeInstanceCount` and `totalPaymentInDollars` at the root level of eqip_ira_summary.json, it will be sufficient to get a national-level total for dollars and instance counts. Thanks.

ywkim312 replied:
@sandeep-ps I removed the total from as state name and removed total as practice name. Also, add  totalPracticeInstanceCount and totalPaymentInDollars of national level at the root elve or summary.json

sandeep-ps replied:
Thank you!

pengyin-shan commented on title-2-conservation/eqip_ira/eqip_ira_state_distribution.json:
```diff
@@ -0,0 +1,16975 @@
+[
+    {
```
Another question about the layout of data structure. Anyway we make the year as the first level (i.e. following what we currently do for title II), then we will have something like:

{
   "2023": [
   "state": "TX"...],
   "2024": ["state":"TX"..]
   ..}

ywkim312 replied:
There are multiple year information that starts from 2023 to 2031 and some of the years are missing in some states. So probably making like this is easier in term of big picture? If you put 2023, 2024 separate, there is also no room to put 2023 total, maybe we can put under 2023 though but it will probably cause another confusion by that.

pengyin-shan replied:
For the years that are missing, we can skip the corresponding states in this year's JSON. For example, "2023" could include zero input if no state has a prediction for this year.

The benefit of putting years as the first level is that Jonathan mentioned he would like to compare year-by-year. Thus, we will need to implement the feature that retrieves a year's data and compare it with another year's data. The request I may send from the front end could be `.../q=ira&year=2023` or `all_ira_data[2023]` (assume all_ira_data is the big json). It would be easier to derive all years for these kinds of requests.

However, I am open to other solutions as well and would like to hear everyone's thoughts.

ywkim312 replied:
Oh okay, then let's change that way

ywkim312 replied:
so let's confirm
you want to make like this

{
"2023": [
"state": "TX"...],
"2024": ["state":"TX"..]
..}

pengyin-shan replied:
Yes, it would be great to have the year as the first level, so something like:

```javascript
{
    "2023": [
      {"state": "TX", totalPayment: xxx, "practices": {...}},
      {"state": "AB", totalPayment: xxx, "practices": {...}},
      ...],
    "2024": [
    {
    {"state": "TX", totalPayment: xxx, "practices": {...}},
    {"state": "AB", totalPayment: xxx, "practices": {...}},
    ...],
    ...
    "2031": [] // assume no state for this year
}
```

Thank you @ywkim312 !

ywkim312 committed:
restruectured state distribution

pengyin-shan commented:
Hi @ywkim312 , the changes are great 👍, thank you very much for your work! 

---

Just a few very minor naming convention changes, if possible:

1. For non-prediction name, can we change `totalPayment` to `totalPaymentInDollars` on the top level? For the sub-payment in *practices*, I noticed that we have `2023PaymentInDollars`, which is equal to the `totalPaymentInDollars`. I suggest we remove the `2023 PaymentInDollars`, as illustrated here:
<img width="459" alt="Screenshot 2024-05-22 at 12 00 46 PM" src="https://github.com/policy-design-lab/data-import/assets/92752107/4fb13b98-9afd-4ddc-9dc0-36489603298c">

By doing this, we will achieve a data structure very similar to the existing EQIP/Title II data structure. This will also lead to unified naming for attributes with the same meaning, allowing us to reuse some front-end components.

---

2. For the predictions, can we change `MinimumDollars` to `minimumTotalPaymentInDollars`, as well as `MaximumDollars` to `maximumTotalPaymentInDollars`? Again, this will make our naming convention unified:
<img width="483" alt="Screenshot 2024-05-22 at 12 08 05 PM" src="https://github.com/policy-design-lab/data-import/assets/92752107/4ef9bb91-cd96-4d41-acc0-2723939935d0">

Lastly, just to confirm for predictions, we don't have a national level total minimum and total maximum number right? (as illustrated above)

ywkim312 committed:
refactor the output state distribution format

ywkim312 commented:
> Hi @ywkim312 , the changes are great 👍, thank you very much for your work!
> 
> Just a few very minor naming convention changes, if possible:
> 
> 1. For non-prediction name, can we change `totalPayment` to `totalPaymentInDollars` on the top level? For the sub-payment in _practices_, I noticed that we have `2023PaymentInDollars`, which is equal to the `totalPaymentInDollars`. I suggest we remove the `2023 PaymentInDollars`, as illustrated here:
> 
> <img alt="Screenshot 2024-05-22 at 12 00 46 PM" width="459" src="https://private-user-images.githubusercontent.com/92752107/332892631-4fb13b98-9afd-4ddc-9dc0-36489603298c.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTY0ODQ3NDUsIm5iZiI6MTcxNjQ4NDQ0NSwicGF0aCI6Ii85Mjc1MjEwNy8zMzI4OTI2MzEtNGZiMTNiOTgtOWFmZC00ZGRjLTlkYzAtMzY0ODk2MDMyOThjLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA1MjMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNTIzVDE3MTQwNVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWU0YjBlOWYzMGQ0ZmZkN2IzOWU2MzViYjk4YjViOTBhNTdmZjdjNTE4ZDU5MWJiMDA4NmE5YWVmOTNmNjY0YTQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.Bzzze4eCcP3B9_JBg-0QB0fnD6rsQqSqKrG4JhNYrWw">
> By doing this, we will achieve a data structure very similar to the existing EQIP/Title II data structure. This will also lead to unified naming for attributes with the same meaning, allowing us to reuse some front-end components.
> 
> 2. For the predictions, can we change `MinimumDollars` to `minimumTotalPaymentInDollars`, as well as `MaximumDollars` to `maximumTotalPaymentInDollars`? Again, this will make our naming convention unified:
> 
> <img alt="Screenshot 2024-05-22 at 12 08 05 PM" width="483" src="https://private-user-images.githubusercontent.com/92752107/332894745-4ef9bb91-cd96-4d41-acc0-2723939935d0.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTY0ODQ3NDUsIm5iZiI6MTcxNjQ4NDQ0NSwicGF0aCI6Ii85Mjc1MjEwNy8zMzI4OTQ3NDUtNGVmOWJiOTEtY2Q5Ni00ZDQxLWFjYzAtMjcyMzkzOTkzNWQwLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA1MjMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNTIzVDE3MTQwNVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTI3ZWU3NmYyNTQ4ZjFjYjgzZDdjYmFhMjJmZDJlNjVkMDIzN2NjZTFlNDJlMDg1OWJhOWJiOTNmZDZiNjcwNDAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.hSj4KGNS2MbRi5didRAVbToIo4F8UL-IwzhXN-o1RXM">
> Lastly, just to confirm for predictions, we don't have a national level total minimum and total maximum number right? (as illustrated above)

Formats has been changed. 

No, we don't have a national level total, but I added the each states total in the output json

pengyin-shan commented:
> > Hi @ywkim312 , the changes are great 👍, thank you very much for your work!
> > Just a few very minor naming convention changes, if possible:
> > 
> > 1. For non-prediction name, can we change `totalPayment` to `totalPaymentInDollars` on the top level? For the sub-payment in _practices_, I noticed that we have `2023PaymentInDollars`, which is equal to the `totalPaymentInDollars`. I suggest we remove the `2023 PaymentInDollars`, as illustrated here:
> > 
> > <img alt="Screenshot 2024-05-22 at 12 00 46 PM" width="459" src="https://private-user-images.githubusercontent.com/92752107/332892631-4fb13b98-9afd-4ddc-9dc0-36489603298c.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTY0ODQ3NDUsIm5iZiI6MTcxNjQ4NDQ0NSwicGF0aCI6Ii85Mjc1MjEwNy8zMzI4OTI2MzEtNGZiMTNiOTgtOWFmZC00ZGRjLTlkYzAtMzY0ODk2MDMyOThjLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA1MjMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNTIzVDE3MTQwNVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWU0YjBlOWYzMGQ0ZmZkN2IzOWU2MzViYjk4YjViOTBhNTdmZjdjNTE4ZDU5MWJiMDA4NmE5YWVmOTNmNjY0YTQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.Bzzze4eCcP3B9_JBg-0QB0fnD6rsQqSqKrG4JhNYrWw">
> > By doing this, we will achieve a data structure very similar to the existing EQIP/Title II data structure. This will also lead to unified naming for attributes with the same meaning, allowing us to reuse some front-end components.
> > 
> > 2. For the predictions, can we change `MinimumDollars` to `minimumTotalPaymentInDollars`, as well as `MaximumDollars` to `maximumTotalPaymentInDollars`? Again, this will make our naming convention unified:
> > 
> > <img alt="Screenshot 2024-05-22 at 12 08 05 PM" width="483" src="https://private-user-images.githubusercontent.com/92752107/332894745-4ef9bb91-cd96-4d41-acc0-2723939935d0.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTY0ODQ3NDUsIm5iZiI6MTcxNjQ4NDQ0NSwicGF0aCI6Ii85Mjc1MjEwNy8zMzI4OTQ3NDUtNGVmOWJiOTEtY2Q5Ni00ZDQxLWFjYzAtMjcyMzkzOTkzNWQwLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA1MjMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNTIzVDE3MTQwNVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTI3ZWU3NmYyNTQ4ZjFjYjgzZDdjYmFhMjJmZDJlNjVkMDIzN2NjZTFlNDJlMDg1OWJhOWJiOTNmZDZiNjcwNDAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.hSj4KGNS2MbRi5didRAVbToIo4F8UL-IwzhXN-o1RXM">
> > Lastly, just to confirm for predictions, we don't have a national level total minimum and total maximum number right? (as illustrated above)
> 
> Formats has been changed.
> 
> No, we don't have a national level total, but I added the each states total in the output json

Hi @ywkim312 ,thanks so much for these changes! It helps a lot to keep our code in a unified style and reusable in the future! 👍 just a extremely small thing: I think there's an extra 'Year' at the end of the `"maximumTotalPaymentInDollarsYear"` attribute:

<img width="483" alt="Screenshot 2024-05-23 at 4 40 10 PM" src="https://github.com/policy-design-lab/data-import/assets/92752107/013af7f5-ebbb-48ed-bd7b-95252c4d6a01">

ywkim312 committed:
fixed typo

ywkim312 commented:
> > > Hi @ywkim312 , the changes are great 👍, thank you very much for your work!
> > > Just a few very minor naming convention changes, if possible:
> > > 
> > > 1. For non-prediction name, can we change `totalPayment` to `totalPaymentInDollars` on the top level? For the sub-payment in _practices_, I noticed that we have `2023PaymentInDollars`, which is equal to the `totalPaymentInDollars`. I suggest we remove the `2023 PaymentInDollars`, as illustrated here:
> > > 
> > > <img alt="Screenshot 2024-05-22 at 12 00 46 PM" width="459" src="https://private-user-images.githubusercontent.com/92752107/332892631-4fb13b98-9afd-4ddc-9dc0-36489603298c.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTY0ODQ3NDUsIm5iZiI6MTcxNjQ4NDQ0NSwicGF0aCI6Ii85Mjc1MjEwNy8zMzI4OTI2MzEtNGZiMTNiOTgtOWFmZC00ZGRjLTlkYzAtMzY0ODk2MDMyOThjLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA1MjMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNTIzVDE3MTQwNVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWU0YjBlOWYzMGQ0ZmZkN2IzOWU2MzViYjk4YjViOTBhNTdmZjdjNTE4ZDU5MWJiMDA4NmE5YWVmOTNmNjY0YTQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.Bzzze4eCcP3B9_JBg-0QB0fnD6rsQqSqKrG4JhNYrWw">
> > > By doing this, we will achieve a data structure very similar to the existing EQIP/Title II data structure. This will also lead to unified naming for attributes with the same meaning, allowing us to reuse some front-end components.
> > > 
> > > 2. For the predictions, can we change `MinimumDollars` to `minimumTotalPaymentInDollars`, as well as `MaximumDollars` to `maximumTotalPaymentInDollars`? Again, this will make our naming convention unified:
> > > 
> > > <img alt="Screenshot 2024-05-22 at 12 08 05 PM" width="483" src="https://private-user-images.githubusercontent.com/92752107/332894745-4ef9bb91-cd96-4d41-acc0-2723939935d0.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTY0ODQ3NDUsIm5iZiI6MTcxNjQ4NDQ0NSwicGF0aCI6Ii85Mjc1MjEwNy8zMzI4OTQ3NDUtNGVmOWJiOTEtY2Q5Ni00ZDQxLWFjYzAtMjcyMzkzOTkzNWQwLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA1MjMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNTIzVDE3MTQwNVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTI3ZWU3NmYyNTQ4ZjFjYjgzZDdjYmFhMjJmZDJlNjVkMDIzN2NjZTFlNDJlMDg1OWJhOWJiOTNmZDZiNjcwNDAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.hSj4KGNS2MbRi5didRAVbToIo4F8UL-IwzhXN-o1RXM">
> > > Lastly, just to confirm for predictions, we don't have a national level total minimum and total maximum number right? (as illustrated above)
> > 
> > 
> > Formats has been changed.
> > No, we don't have a national level total, but I added the each states total in the output json
> 
> Hi @ywkim312 ,thanks so much for these changes! It helps a lot to keep our code in a unified style and reusable in the future! 👍 just a extremely small thing: I think there's an extra 'Year' at the end of the `"maximumTotalPaymentInDollarsYear"` attribute:
> 
> <img alt="Screenshot 2024-05-23 at 4 40 10 PM" width="483" src="https://private-user-images.githubusercontent.com/92752107/333349016-013af7f5-ebbb-48ed-bd7b-95252c4d6a01.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTY1MDExMjYsIm5iZiI6MTcxNjUwMDgyNiwicGF0aCI6Ii85Mjc1MjEwNy8zMzMzNDkwMTYtMDEzYWY3ZjUtZWJiYi00OGVkLWJkN2ItOTUyNTJjNGQ2YTAxLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA1MjMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNTIzVDIxNDcwNlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTkwM2QzZmFmZmUwN2I0ZTJhOGUxYzZmNjRhMDc0NjU4YzVkOWRkMjY1OTA1YjU0YzY1NTllZGMyYzhmNTAzYjkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.qApsTVzAsVulGoAxqg0Nj4Nk7IpXeZ9S7aVnbOZ8rqI">

Oops, I am sorry. Fixed it

pengyin-shan reviewed (approved):
Looks great, approving this PR. Will start the front-end development based on this data for IRA EQIP.

sandeep-ps commented:
Hi, @ywkim312, I reviewed this PR by comparing the JSON data against Pivot tables generated from the raw CSV files. 

I'm getting the year-wise total as different. For example, for CA, for 2024, minimumTotalPaymentInDollars comes to $18,226,729.12 in my calculation, but in the JSON file, it's $17,413,510.35. Could you please check this again? I'll share the pivot table via Slack.

A few other thoughts, but they may not be of high priority: 

1. It may be better to use `predicted...` prefix to differentiate between predicted vs actual data (current year) 
2. This is not a high priority either, but sorting the list of practices based on the code may be good too.

sandeep-ps commented:
[20231106-2024_2031-EQIPextrafund-project-by-practice-MIN-clean.xlsx](https://github.com/policy-design-lab/data-import/files/15486456/20231106-2024_2031-EQIPextrafund-project-by-practice-MIN-clean.xlsx)

Or maybe it's better to add it here.

ywkim312 commented:
> Hi, @ywkim312, I reviewed this PR by comparing the JSON data against Pivot tables generated from the raw CSV files.
> 
> I'm getting the year-wise total as different. For example, for CA, for 2024, minimumTotalPaymentInDollars comes to $18,226,729.12 in my calculation, but in the JSON file, it's $17,413,510.35. Could you please check this again? I'll share the pivot table via Slack.
> 
> A few other thoughts, but they may not be of high priority:
> 
> 1. It may be better to use `predicted...` prefix to differentiate between predicted vs actual data (current year)
> 2. This is not a high priority either, but sorting the list of practices based on the code may be good too

For 2024 total, not all the columns in the xls files are included. It only has the matching values with the practice number. Therefore, some columns will be missed because it can't matched to the ones in the 2023 table. So It kind of make sense that my number is lower than yours.

Also, can you make more detailed explain your suggestion number 1?

ywkim312 commented:
> [20231106-2024_2031-EQIPextrafund-project-by-practice-MIN-clean.xlsx](https://github.com/policy-design-lab/data-import/files/15486456/20231106-2024_2031-EQIPextrafund-project-by-practice-MIN-clean.xlsx)
> 
> Or maybe it's better to add it here.

Still not very clear what you mean. I am sorry to ask but can you make little more in detial?

sandeep-ps commented:
Let me look at the total values again, and I will get back to you soon.

Regarding 

>It may be better to use predicted... prefix to differentiate between predicted vs actual data (current year)

I meant since `minimumTotalPaymentInDollars` and `maximumTotalPaymentInDollars` for years 2024-2031 are "predicted" payments, it may be better to say `predictedMinimumTotalPaymentInDollars` and `predictedMaximumTotalPaymentInDollars`.

ywkim312 commented:
> Let me look at the total values again, and I will get back to you soon.
> 
> Regarding
> 
> > It may be better to use predicted... prefix to differentiate between predicted vs actual data (current year)
> 
> I meant since `minimumTotalPaymentInDollars` and `maximumTotalPaymentInDollars` for years 2024-2031 are "predicted" payments, it may be better to say `predictedMinimumTotalPaymentInDollars` and `predictedMaximumTotalPaymentInDollars`.

Oh I see. Thank you for the answer @pengyin-shan How do you think?

pengyin-shan commented:
> > Let me look at the total values again, and I will get back to you soon.
> > Regarding
> > > It may be better to use predicted... prefix to differentiate between predicted vs actual data (current year)
> > 
> > 
> > I meant since `minimumTotalPaymentInDollars` and `maximumTotalPaymentInDollars` for years 2024-2031 are "predicted" payments, it may be better to say `predictedMinimumTotalPaymentInDollars` and `predictedMaximumTotalPaymentInDollars`.
> 
> Oh I see. Thank you for the answer @pengyin-shan How do you think?

Yes, this will work for me. Please feel free to add 'predicted' prefixes for all predicted values, and I'll refer to this prefix for all future 'predicted' data.

ywkim312 committed:
Updated total nationwide values in summary

ywkim312 commented:
> > > Let me look at the total values again, and I will get back to you soon.
> > > Regarding
> > > > It may be better to use predicted... prefix to differentiate between predicted vs actual data (current year)
> > > 
> > > 
> > > I meant since `minimumTotalPaymentInDollars` and `maximumTotalPaymentInDollars` for years 2024-2031 are "predicted" payments, it may be better to say `predictedMinimumTotalPaymentInDollars` and `predictedMaximumTotalPaymentInDollars`.
> > 
> > 
> > Oh I see. Thank you for the answer @pengyin-shan How do you think?
> 
> Yes, this will work for me. Please feel free to add 'predicted' prefixes for all predicted values, and I'll refer to this prefix for all future 'predicted' data.

@sandeep-ps @pengyin-shan I changed to predictedMinimum and predictedMaximum

ywkim312 committed:
sort the summary by practice name

sandeep-ps commented:
Thanks, Yongwook! The changes look good to me. I will approve this PR.

sandeep-ps reviewed (approved):
Approving. Thanks.
